app/services/hts_service/reports/hts_report_builder.rb

```diff
@@ -38,115 +38,94 @@ def his_patients_rev
             )
         end
 
-  def his_patients_revs(indicators)
-
-            columns = ActiveRecord::Base.connection.columns('obs').map(&:name) 
-          sql_query = <<~SQL
-            SELECT patients.patient_id, person.gender, person.birthdate, encounter.encounter_datetime,
-                   CONCAT_WS('/', #{columns.map { |col| "COALESCE(obs.#{col}, 'NULL')" }.join(', ')}) AS observations
-            FROM (
-                SELECT patient.patient_id
-                FROM patient
-                INNER JOIN person ON person.person_id = patient.patient_id AND patient.voided = 0 
-                INNER JOIN encounter ON encounter.patient_id = patient.patient_id
-                INNER JOIN program ON program.program_id = encounter.program_id
-                WHERE encounter.encounter_datetime BETWEEN '#{@start_date}' AND '#{@end_date}'
-                AND encounter.encounter_type = (
-                    SELECT encounter_type_id FROM encounter_type WHERE name = 'Testing'
-                )
-                AND program.program_id = (
-                    SELECT program_id FROM program WHERE name = 'HTC PROGRAM'
-                )
-            ) AS patients
-            INNER JOIN obs ON patients.patient_id = obs.person_id AND obs.voided = 0
-            INNER JOIN person ON person.person_id = patients.patient_id
-            INNER JOIN encounter ON encounter.patient_id = patients.patient_id;
-          SQL
-        
-          results = ActiveRecord::Base.connection.execute(sql_query)
-        
-          data = {}
-        
-          results.each do |row|
-            patient_id = row[0]
-            observations = row[4].split('/')
-            patient_obs = {}
-        
-            columns.each_with_index do |value, index|
-              patient_obs[value] = observations[index]
-            end
-        
-            patient_data = data[patient_id]
-            if patient_data.nil?
-              patient_data = { "patientid" => patient_id, "observations" => [], "demographics" => { "gender" => row[1], "person_id" => patient_id, "birthdate" => row[2], "encounter_datetime" => row[3] } }
-              data[patient_id] = patient_data
-            end
-        
-            patient_data["observations"] << patient_obs
-          end
-        
-          return process_patient_data(data, indicators)
-    end
-        
-    def process_patient_data(data, indicators)
-          patient_data = []
+      def his_patients_revs(indicators) 
+    
+                columns = ActiveRecord::Base.connection.columns('obs').map(&:name) 
+                sql_query = <<~SQL
+                SELECT
+                person.gender,
+                person.birthdate,
+                encounter.encounter_datetime,
+                obs.*
+              FROM patient
+              INNER JOIN person ON person.person_id = patient.patient_id
+              INNER JOIN encounter ON encounter.patient_id = patient.patient_id
```

Comment:
```suggestion
INNER JOIN encounter ON encounter.patient_id = patient.patient_id AND encounter.voided = 0
```
the filters in the where clause should also come here